### ISSUE TITLE
Inform AI of host OS for execute_shell

### ISSUE THREAD

**Issue by nitefliteuk** [Significant-Gravitas/Auto-GPT#2390]
### ⚠️ Search for existing issues first ⚠️

- [X] I have searched the existing issues, and there is no existing issue for my problem

### GPT-3 or GPT-4

- [ ] I am using Auto-GPT with GPT-3 (GPT-3.5)

### Steps to reproduce 🕹

This issue relates and expands issue #2257 

Configuration: 
- AutoGPT stable version v0.2.1
- Windows 10 Pro 22H2
- miniconda with Python 3.10.9
- all requirements installed
- full prompt provided below

### Current behavior 😯

AutoGPT doesn't check the host OS on startup and tries to execute Linux bash commands on a Windows machine. That includes using the wrong path separator (as reported in issue #2257). In doing so it's unable to create nested subfolders ('projectname\data') and at times but not always, unable to save python files with the correct line termination.

Example line below:
NEXT ACTION:  COMMAND = execute_shell ARGUMENTS = {'command_line': 'mkdir -p wine-quality/wget && cd wine-quality/wget && wget https://ftp.gnu.org/gnu/wget/wget-1.21.2.tar.gz && tar -xvf wget-1.21.2.tar.gz && cd wget-1.21.2 && ./configure && make && make install && echo "export PATH=$PATH:/usr/local/bin" >> ~/.bashrc && source ~/.bashrc'}

### Expected behavior 🤔

Be aware of the host OS and use compatible syntax and commands.

### Your prompt 📝

ai_goals:
- Create the working directory 'wine-quality', download any .csv dataset from https://archive.ics.uci.edu/ml/machine-learning-databases/wine-quality/ and save them in the folder 'wine-quality\data'
- Import the csv datasets and use the last column as output
- Develop the most accurate machine learning model using TensorFlow and Keras to maximise the
  output accuracy. Any model training and testing should be executed locally and not using cloud-based virtual machines
- Write a Jupyter notebook including all code to import datasets, set up the best model
  found in previous step and presenting results both numerically and graphically.
  The jupyter notebook should be saved under the folder 'wine-quality\output'
ai_name: Wine tester
ai_role: An AI to autonomously improve wine detection results from a well-known dataset.
  Any python file required to achieve the goal should be saved in the folder 'wine-quality'
  and any dataset stored locally under the folder 'wine-quality\data'

===================== Log ===============================
2023-04-18 15:09:22,136 INFO Welcome back!  Would you like me to return to being Wine tester?
2023-04-18 15:09:25,586 INFO Using memory of type: LocalCache
2023-04-18 15:09:25,809 INFO Using Browser: chrome
2023-04-18 15:09:41,832 INFO  THOUGHTS: I could start by browsing the website https://archive.ics.uci.edu/ml/machine-learning-databases/wine-quality/ to find a suitable dataset
2023-04-18 15:09:42,263 INFO REASONING: Browsing the website may give me some insight on how to obtain the right dataset for this task
2023-04-18 15:09:42,914 INFO PLAN: 
2023-04-18 15:09:42,976 INFO -  Browse the website https://archive.ics.uci.edu/ml/machine-learning-databases/wine-quality/ for appropriate dataset
2023-04-18 15:09:43,253 INFO -  After obtaining a suitable dataset, I can proceed with the rest of the tasks
2023-04-18 15:09:43,637 INFO CRITICISM: I need to be careful on the dataset to be chosen to ensure it has the right inputs and outputs as required
2023-04-18 15:09:44,239 INFO SPEAK: I am planning to browse the website https://archive.ics.uci.edu/ml/machine-learning-databases/wine-quality/ to find a suitable dataset for this task.
2023-04-18 15:09:44,732 INFO Attempting to fix JSON by finding outermost brackets
 
2023-04-18 15:09:45,073 INFO Apparently json was fixed. 
2023-04-18 15:09:45,289 INFO NEXT ACTION:  COMMAND = browse_website  ARGUMENTS = {'url': 'https://archive.ics.uci.edu/ml/machine-learning-databases/wine-quality/', 'question': 'How to obtain the CSV dataset from the website?'}
2023-04-18 15:09:49,904 INFO -=-=-=-=-=-=-= COMMAND AUTHORISED BY USER -=-=-=-=-=-=-= 
2023-04-18 15:10:05,517 INFO SYSTEM:  Command browse_website returned: ('Answer gathered from website: To obtain the CSV dataset from the website, you need to navigate to the "wine-quality" directory and select either "winequality-red.csv" or "winequality-white.csv." \n \n Links: [\' Parent Directory (https://archive.ics.uci.edu/ml/machine-learning-databases/)\', \' winequality-red.csv (https://archive.ics.uci.edu/ml/machine-learning-databases/wine-quality/winequality-red.csv)\', \' winequality-white.csv (https://archive.ics.uci.edu/ml/machine-learning-databases/wine-quality/winequality-white.csv)\', \' winequality.names (https://archive.ics.uci.edu/ml/machine-learning-databases/wine-quality/winequality.names)\']', <selenium.webdriver.chrome.webdriver.WebDriver (session="98f2c71fd32b223aeff98af17ddc035b")>)
2023-04-18 15:10:19,973 INFO  THOUGHTS: I will now download csv dataset by executing a shell command
2023-04-18 15:10:20,359 INFO REASONING: Executing a shell command is a quick way of downloading the dataset
2023-04-18 15:10:20,785 INFO PLAN: 
2023-04-18 15:10:20,847 INFO -  Download the csv dataset by executing a shell command in the command line
2023-04-18 15:10:21,324 INFO CRITICISM: There is a possibility that the download might stall or fail completely
2023-04-18 15:10:21,709 INFO SPEAK: I will now download the dataset by executing a shell command in the command line.
2023-04-18 15:10:22,168 INFO Attempting to fix JSON by finding outermost brackets
 
2023-04-18 15:10:22,474 INFO Apparently json was fixed. 
2023-04-18 15:10:22,660 INFO NEXT ACTION:  COMMAND = execute_shell  ARGUMENTS = {'command_line': 'mkdir -p wine-quality/data && cd wine-quality/data && wget https://archive.ics.uci.edu/ml/machine-learning-databases/wine-quality/winequality-white.csv && wget https://archive.ics.uci.edu/ml/machine-learning-databases/wine-quality/winequality-red.csv'}
2023-04-18 15:10:29,703 INFO -=-=-=-=-=-=-= COMMAND AUTHORISED BY USER -=-=-=-=-=-=-= 
2023-04-18 15:10:30,311 INFO SYSTEM:  Command execute_shell returned: STDOUT:
b''
STDERR:
b'The syntax of the command is incorrect.\r\n'
2023-04-18 15:10:50,764 INFO  THOUGHTS: I need to modify the execute_shell command to fix syntax issues. I will use strings rather than quotes to execute multi-line command. Installing wget command will also be better for downloading files
2023-04-18 15:10:51,537 INFO REASONING: Using a string for the command separates each command, which makes it simpler to understand and identify errors. Installing wget using the command line will ensure efficient downloading of the files
2023-04-18 15:10:52,280 INFO PLAN: 
2023-04-18 15:10:52,342 INFO -  Use a string with escape character '' instead of quotes to separate command lines in the 'execute_shell' command
2023-04-18 15:10:52,893 INFO -  Start by installing wget command to ensure it is present
2023-04-18 15:10:53,278 INFO -  After installing wget, execute commands to create the directories and download csv datasets
2023-04-18 15:10:53,696 INFO CRITICISM: There is still a possibility that there may be syntax issues if the command is not entered correctly
2023-04-18 15:10:54,293 INFO SPEAK: I need to modify the execute_shell command by using strings with escape character '\' instead of quotes. I also need to install the wget command to download files efficiently and in the end I will execute commands to create directories and download datasets.
2023-04-18 15:10:55,269 INFO Attempting to fix JSON by finding outermost brackets
 
2023-04-18 15:10:55,564 INFO Apparently json was fixed. 
2023-04-18 15:10:55,717 INFO NEXT ACTION:  COMMAND = execute_shell  ARGUMENTS = {'command_line': 'apt-get install wget -y && mkdir -p wine-quality/data && cd wine-quality/data && wget https://archive.ics.uci.edu/ml/machine-learning-databases/wine-quality/winequality-white.csv && wget https://archive.ics.uci.edu/ml/machine-learning-databases/wine-quality/winequality-red.csv'}
2023-04-18 15:11:00,483 INFO -=-=-=-=-=-=-= COMMAND AUTHORISED BY USER -=-=-=-=-=-=-= 
2023-04-18 15:11:01,168 INFO SYSTEM:  Command execute_shell returned: STDOUT:
b''
STDERR:
b"'apt-get' is not recognized as an internal or external command,\r\noperable program or batch file.\r\n"
2023-04-18 15:11:19,592 INFO  THOUGHTS: The previous command did not execute successfully because the environment being used currently cannot recognize the apt-get package manager. Instead of installing the wget package, I will download it from the website and install it to download csv datasets.
2023-04-18 15:11:20,427 INFO REASONING: The package manager cannot be run due to the environment being used. Hence, downloading the package from the website can be executed effectively
2023-04-18 15:11:21,045 INFO PLAN: 
2023-04-18 15:11:21,121 INFO -  Download wget package by executing a command in the command line
2023-04-18 15:11:21,480 INFO -  After downloading the package, install it to allow downloading csv datasets
2023-04-18 15:11:21,847 INFO CRITICISM: There may be problems with the download or installation which could cause errors
2023-04-18 15:11:22,278 INFO SPEAK: I need to download the wget package to enable me to download csv datasets because my current environment cannot recognize the apt-get package manager
2023-04-18 15:11:22,927 INFO Attempting to fix JSON by finding outermost brackets
 
2023-04-18 15:11:23,201 INFO Apparently json was fixed. 
2023-04-18 15:11:23,385 INFO NEXT ACTION:  COMMAND = execute_shell  ARGUMENTS = {'command_line': 'mkdir -p wine-quality/wget && cd wine-quality/wget && wget https://ftp.gnu.org/gnu/wget/wget-1.21.2.tar.gz && tar -xvf wget-1.21.2.tar.gz && cd wget-1.21.2 && ./configure && make && make install && echo "export PATH=$PATH:/usr/local/bin" >> ~/.bashrc && source ~/.bashrc'}
2023-04-18 15:11:28,275 INFO -=-=-=-=-=-=-= COMMAND AUTHORISED BY USER -=-=-=-=-=-=-= 
2023-04-18 15:11:28,742 INFO SYSTEM:  Command execute_shell returned: STDOUT:
b''
STDERR:
b'The syntax of the command is incorrect.\r\n'
2023-04-18 15:11:51,690 INFO  THOUGHTS: I need to modify the string passed to the execute shell command to fix the syntax error in the previous command. I also need to test if wget is installed correctly by running the command 'wget --version' in the command line
2023-04-18 15:11:52,493 INFO REASONING: Correcting any syntax errors in the execute shell command will allow the command to execute successfully. Testing for the installation of wget command is necessary before using it for dataset download
2023-04-18 15:11:53,238 INFO PLAN: 
2023-04-18 15:11:53,299 INFO -  Use strings with escape character '\' instead of quotes
2023-04-18 15:11:53,605 INFO -  Test for the installation of wget command by executing the command 'wget --version' in the command line
2023-04-18 15:11:54,157 INFO CRITICISM: There is a possibility that there might be issues with syntax if the string is not entered correctly. Testing for the installation of wget is not a sure-fire way of knowing if wget will work correctly for dataset download as it may encounter problems later during the process
2023-04-18 15:11:55,187 INFO SPEAK: I need to modify the string passed to the execute shell command to fix the syntax error to ensure I can execute shell commands properly. I should also test if wget installed correctly by running the command 'wget --version'.
2023-04-18 15:11:56,050 INFO Attempting to fix JSON by finding outermost brackets
 
2023-04-18 15:11:56,313 INFO Apparently json was fixed. 
2023-04-18 15:11:56,515 INFO NEXT ACTION:  COMMAND = execute_shell  ARGUMENTS = {'command_line': 'wget --version && mkdir -p wine-quality/data && cd wine-quality/data && wget https://archive.ics.uci.edu/ml/machine-learning-databases/wine-quality/winequality-white.csv && wget https://archive.ics.uci.edu/ml/machine-learning-databases/wine-quality/winequality-red.csv'}
2023-04-18 15:12:23,746 INFO SYSTEM:  Human feedback: please take into account that this is a Windows machine
2023-04-18 15:12:31,225 WARNING  Warning: Failed to parse AI output, attempting to fix.
 If you see this warning frequently, it's likely that your prompt is confusing the AI. Try changing it up slightly.
2023-04-18 15:12:48,358 INFO  THOUGHTS: The command downloads two csv files for wine quality data and stores them in a newly created directory called 'wine-quality/data'.
2023-04-18 15:12:48,911 INFO REASONING: The command uses the 'wget' command to download the csv files from the given URLs. The 'mkdir' command is used to create a new directory called 'wine-quality/data' to store the downloaded files. The 'cd' command is used to navigate to the newly created directory. The '&&' operator is used to chain multiple commands together.
2023-04-18 15:12:50,020 INFO PLAN: 
2023-04-18 15:12:50,096 INFO -  The command could be improved by adding error handling in case the download fails or the directory cannot be created.
2023-04-18 15:12:50,696 INFO CRITICISM: 
2023-04-18 15:12:50,774 INFO SPEAK: I have executed the command to download the wine quality data csv files and store them in a new directory called 'wine-quality/data'.
2023-04-18 15:12:51,468 INFO Attempting to fix JSON by finding outermost brackets
 
2023-04-18 15:12:51,730 ERROR Error: Invalid JSON, setting it to empty JSON now.
 
2023-04-18 15:12:51,733 INFO NEXT ACTION:  COMMAND = Error:  ARGUMENTS = 'dict' object has no attribute 'replace'
